task-tracker-console-application/src/main/java/com/isa/user/Login.java

```diff
@@ -0,0 +1,135 @@
+        inputLogin = scanner.nextLine();
+        if (inputLogin.equals("0")) {
+            goToMainMenu();
+        } else if (chosenUser().isEmpty()) {
+            incorrectLoginInputCount++;
+            System.out.println("Podany login nie istnieje.");
+            if (incorrectLoginInputCount > 1) {
+                System.out.println("Pamiętaj, że w każdej chwili możesz wrócić do menu głównego poprzez wybór \"0\".");
+            }
+
+            System.out.print("Podaj login ponownie: ");
+            validateLogin(scanner, incorrectLoginInputCount);
+        } else {
+            System.out.print("Podaj hasło: ");
+            validatePassword(scanner, 0);
+        }
+    }
+
+    private static Optional<User> chosenUser() {
+        return Optional.ofNullable(users.stream().
+                filter(user -> user.getLogin().equals(inputLogin)).findFirst().orElse(null));
+    }
+
+    private static void validatePassword(Scanner scanner, int incorrectPasswordInputCount) {
+        String inputPassword = scanner.nextLine();
+        if (inputPassword.equals("0")) {
+            goToMainMenu();
+        } else if (!isCorrectPassword(inputPassword)) {
+            checkForIncorrectPassword(scanner, incorrectPasswordInputCount);
+        } else {
+            loggedInUser = chosenUser().get();
+            printWelcomeMessage();
+            if (loggedInUser.isActive()) {
+                /* TODO Replace the message below once the menu for the logged in user is created */
+                System.out.println("Menu po zalogowaniu się niedostępne. Nastąpi samoczynne zakończenie programu.");
+            } else {
+                askForBlockedUserChoice(scanner);
+            }
+        }
+    }
+
+    private static void checkForIncorrectPassword(Scanner scanner, int incorrectPasswordInputCount) {
+        incorrectPasswordInputCount++;
+        System.out.println("Hasło niepoprawne.");
+        if (incorrectPasswordInputCount > 1 && incorrectPasswordInputCount < 5) {
+            System.out.println("Pamiętaj, że w każdej chwili możesz wrócić do menu głównego poprzez wybór \"0\".");
+            if (incorrectPasswordInputCount == 4) {
+                System.out.println("UWAGA: Przy kolejnej nieprawidłowej próbie zalogowania się twoje konto zostanie zablokowane.");
+            }
+        }
+        if (incorrectPasswordInputCount != 5) {
+            System.out.print("Podaj hasło ponownie: ");
+            validatePassword(scanner, incorrectPasswordInputCount);
+        } else {
+            chosenUser().ifPresent(user -> user.setActive(false));
+            System.out.println("Wykryto 5 nieprawidłowych prób zalogowania się. Twoje konto zostało zablokowane.");
+            askForBlockedUserChoice(scanner);
+        }
+    }
+
+    private static boolean isCorrectPassword(String inputPassword) {
+        return inputPassword.equals(chosenUser().get().getPassword());
+    }
```

Comment:
a może zamiast tego geta, który w przypadku jak optional będzie pusty rzuci nam NPE to byśmy napisali coś takiego:
```
private static boolean isCorrectPassword(String inputPassword) {
    return chosenUser()
        .map(user -> inputPassword.equals(user.getPassword()))
        .orElse(false);
}
```

Reply:
Poprawione zgodnie z sugestią; dzięki!